karapace/schema_reader.py

```diff
@@ -364,7 +364,13 @@ def consume_messages(self, msgs: list[Message], watch_offsets: bool) -> None:
                 assert message_key is not None
                 key = json_decode(message_key)
             except JSONDecodeError as exc:
-                LOG.warning("Invalid JSON in msg.key() at offset %s", msg.offset())
+                non_bytes_key = msg.key().decode()  # type: ignore[union-attr]
+                LOG.warning("Invalid JSON in msg.key(): %s at offset %s", non_bytes_key, msg.offset())
+                self.offset = msg.offset()  # Invalid entry shall also move the offset so Karapace makes progress.
+                self.kafka_error_handler.handle_error(location=KafkaErrorLocation.SCHEMA_READER, error=exc)
+                continue  # [non-strict mode]
+            except AssertionError as exc:
```

Comment:
I wonder if it's possible to be more precise in catching exactly the assert on the `message_key` and not maybe others potentially raised in `json_decode()`

Reply:
Yes, I can move this catch above the json error catch.

Reply:
Maybe with an if statement instead of assert/catch:
```python
        if message_key is None:
                LOG.warning("Empty msg.key() at offset %s", msg.offset())
                self.offset = msg.offset()  # Invalid entry shall also move the offset so Karapace makes progress.
                self.kafka_error_handler.handle_error(location=KafkaErrorLocation.SCHEMA_READER, error=exc)
                continue  # [non-strict mode]

```

Reply:
I get the idea, but I do not want to change the body of the try at this point, I am also not sure why `json.loads()` might raise an assertion error, I mean it's possible, but ideally a component should catch and parse all external facing errors.